### DISCUSSION TITLE
How to make our own custom module

### DISCUSSION THREAD

**Discussion by asilmertkan** [serenity-is/Serenity#6367] [Serenity Ideas]
I want to make my module like as Northwind, Basic Samples eg. I dont know how to do it?

A guide on this would be great.

**Comment by wirble** [comment]
I would recommend you through the tutorial and documents first.  It will answer 90% of your questions.  The stuff the documents don't tell you are in the discussions and issues.  The last 10% are usually more advanced functions...
[tutorial](https://serenity.is/docs/tutorials/movies/movies)

**Reply by stixoffire** [reply]
Like if all else fails before asking question read documentation and view tutorial .. 
If that does not work try to figure it out, then if that is not working, ask question.

**Comment by asilmertkan** [comment]
Hey @wirble,

Please correct me if I'm missing something. I'm pretty much in control of most of the tutorial. But I couldn't find any tutorial for make seperate modul. I want to make different custom modules. For example make Todo Module as seperate project like Northwind and I want to add my projects easly when I need this module. Is there anyone who has accomplished this or can guide me on this?

**Reply by stixoffire** [reply]
Movies DB is separate module. Did you not read the tutorial?

**Comment by dfaruque** [comment]
you can explore https://github.com/serenity-is/common-features
